src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/ConsumerLockContext.cs

```diff
@@ -10,79 +10,80 @@ namespace MassTransit.KafkaIntegration
         readonly IHostConfiguration _hostConfiguration;
         readonly ReceiveSettings _receiveSettings;
+        readonly ChannelExecutorPool<TopicPartition> _channelExecutorPool;
 
-        public ConsumerLockContext(IHostConfiguration hostConfiguration, ReceiveSettings receiveSettings)
+        public ConsumerLockContext(IHostConfiguration hostConfiguration, ReceiveSettings receiveSettings,
+            ChannelExecutorPool<TopicPartition> channelExecutorPool)
         {
             _hostConfiguration = hostConfiguration;
             _receiveSettings = receiveSettings;
+            _channelExecutorPool = channelExecutorPool;
         }
 
-        public async Task Pending(ConsumeResult<TKey, TValue> result)
+        public async Task Pending(ConsumeResult<byte[], byte[]> result)
         {
             LogContext.SetCurrentIfNull(_hostConfiguration.ReceiveLogContext);
 
-            if (_data.TryGetValue(result.Partition, out var data))
+            if (_data.TryGetValue(result.TopicPartition, out var data))
                 await data.Pending(result).ConfigureAwait(false);
         }
 
-        public Task Complete(ConsumeResult<TKey, TValue> result)
+        public Task Complete(ConsumeResult<byte[], byte[]> result)
         {
             LogContext.SetCurrentIfNull(_hostConfiguration.ReceiveLogContext);
 
-            if (_data.TryGetValue(result.Partition, out var data))
+            if (_data.TryGetValue(result.TopicPartition, out var data))
                 data.Complete(result);
 
             return Task.CompletedTask;
         }
 
-        public Task Faulted(ConsumeResult<TKey, TValue> result, Exception exception)
+        public Task Faulted(ConsumeResult<byte[], byte[]> result, Exception exception)
         {
             LogContext.SetCurrentIfNull(_hostConfiguration.ReceiveLogContext);
 
-            if (_data.TryGetValue(result.Partition, out var data))
+            if (_data.TryGetValue(result.TopicPartition, out var data))
                 data.Faulted(result, exception);
 
             return Task.CompletedTask;
         }
 
-        public void OnAssigned(IConsumer<TKey, TValue> consumer, IEnumerable<TopicPartition> partitions)
+        public void OnAssigned(IConsumer<byte[], byte[]> consumer, IEnumerable<TopicPartition> partitions)
         {
             LogContext.SetCurrentIfNull(_hostConfiguration.ReceiveLogContext);
 
             foreach (var partition in partitions)
             {
-                if (_data.TryAdd(partition.Partition, p => new PartitionCheckpointData(partition, consumer, _receiveSettings)))
-                    LogContext.Info?.Log("Partition: {PartitionId} was assigned", partition);
+                if (!_data.TryAdd(partition, p => new PartitionCheckpointData(partition, consumer, _receiveSettings)))
+                    continue;
+
+                _channelExecutorPool.TryAdd(partition);
+                LogContext.Info?.Log("Partition: {PartitionId} was assigned", partition);
```

Comment:
I think these should be debug instead of info, to avoid log spam. Or maybe I'm wrong, but it seems like a lot of extra messages in the log.